src/builtins/BuiltinTypedArray.cpp

```diff
@@ -485,6 +485,100 @@ static Value builtinTypedArrayCopyWithin(ExecutionState& state, Value thisValue,
     return O;
 }
 
+template <const bool isForward>
+static Value fastTypedArrayIndexSearch(TypedArrayObject* arr, size_t k, size_t len, const Value& value)
+{
+    auto type = arr->typedArrayType();
+    if (type == TypedArrayType::BigInt64 || type == TypedArrayType::BigUint64) {
+        if (!value.isBigInt()) {
+            return Value(-1);
+        }
+    } else {
+        if (!value.isNumber()) {
```

Comment:
What about merging `else` and `if` to one `else if` statement here?
And if these conditions rarely happen, adding `UNLIKELY` would be good too.

Reply:
Good point!